src/seeds.proposals.cpp

```diff
@@ -1271,3 +1295,131 @@ ACTION proposals::initnumprop() {
+  check_voice_scope(scope);
+
+  voice_tables voice(get_self(), scope.value);
+  auto vitr = voice.find(delegator.value);
+  check(vitr != voice.end(), "delegatee does not have voice");
+
+  delegate_trust_tables deltrusts(get_self(), scope.value);
+  auto ditr = deltrusts.find(delegator.value);
+
+  if (ditr != deltrusts.end()) {
+    deltrusts.modify(ditr, _self, [&](auto & item){
+      item.delegatee = delegatee;
+      item.weight = 1.0;
+      item.timestamp = eosio::current_time_point().sec_since_epoch();
+    });
+  } else {
+    deltrusts.emplace(_self, [&](auto & item){
+      item.delegator = delegator;
+      item.delegatee = delegatee;
+      item.weight = 1.0;
+      item.timestamp = eosio::current_time_point().sec_since_epoch();
+    });
+  }
+
+}
+
+void proposals::send_mimic_delegatee_vote (name delegatee, name scope, uint64_t proposal_id, double percentage_used, name option) {
+
+  uint64_t batch_size = config_get("batchsize"_n);
+
+  delegate_trust_tables deltrusts(get_self(), scope.value);
+  auto deltrusts_by_delegatee = deltrusts.get_index<"bydelegatee"_n>();
+
+  auto ditr = deltrusts_by_delegatee.find(delegatee.value);
+
+  if (ditr != deltrusts_by_delegatee.end()) {
+    action mimic_action(
+      permission_level{get_self(), "active"_n},
+      get_self(),
+      "mimicvote"_n,
+      std::make_tuple(delegatee, scope, proposal_id, percentage_used, option, batch_size)
+    );
+
+    transaction tx;
+    tx.actions.emplace_back(mimic_action);
+    tx.delay_sec = 1;
+    tx.send(delegatee.value + 1, _self);
+  }
+
+}
+
+ACTION proposals::mimicvote (name delegatee, name scope, uint64_t proposal_id, double percentage_used, name option, uint64_t chunksize) {
+
+  require_auth(get_self());
+
+  delegate_trust_tables deltrusts(get_self(), scope.value);
+  auto deltrusts_by_delegatee = deltrusts.get_index<"bydelegatee"_n>();
+
+  check_voice_scope(scope);
+  voice_tables voices(get_self(), scope.value);
+
+  auto ditr = deltrusts_by_delegatee.find(delegatee.value);
+  uint64_t count = 0;
+
+  while (ditr != deltrusts_by_delegatee.end() && ditr -> delegatee == delegatee && count < chunksize) {
+
+    name voter = ditr -> delegator;
+    auto vitr = voices.find(voter.value);
+
+    if (option == trust) {
+      vote_aux(voter, proposal_id, vitr -> balance * percentage_used, trust, true, true);
+    } else if (option == distrust) {
+      bool vote_reverted = revert_vote(voter, proposal_id);
+      vote_aux(voter, proposal_id, vitr -> balance * percentage_used, distrust, !vote_reverted, true);
+    } else if (option == abstain) {
+      vote_aux(voter, proposal_id, uint64_t(0), abstain, true, true);
+    }
+
+    ditr++;
+    count++;
+  }
+
+  if (ditr != deltrusts_by_delegatee.end() && ditr -> delegatee == delegatee) {
+    action next_execution(
+      permission_level{get_self(), "active"_n},
+      get_self(),
+      "mimicvote"_n,
+      std::make_tuple(delegatee, scope, proposal_id, percentage_used, option, chunksize)
```

Comment:
Will this not loop forever when size > chunk size? 

The second call signature is the exact same as the first, no pointers are moving. That's a lot of votes..

Reply:
Suggest to add a test with 2 delegations and chunk size 1

Reply:
You are completely right, what a horrible mistake I made here. I fixed in the commit